addons/website_forum/populate/mail_message.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.tools import populate
+
+CP_WEIGHTS = {1: 35, 2: 30, 3: 25, 4: 10}
+
+class Message(models.Model):
+    _inherit = 'mail.message'
+    _populate_dependencies = ['forum.post', 'res.partner']
+    _populate_sizes = {'small': 100, 'medium': 1000, 'large': 90000}
+
+    def _populate_factories(self):
+        mail_message_subtype = self.env.ref('mail.mt_comment')
+
+        def get_res_id(iterator, *args):
+            """Randomly assign messages to some populated posts and answers.
+            This makes sure these questions and answers get one to four comments.
+            """
+            question_ids = self.env.registry.populated_models['forum.post']
+            answers = self.env['forum.post'].search_read([('parent_id', 'in', question_ids)], ['id'])
```

Comment:
Always a bit lost with generators... how many times will this be called ? Just to be sure we don't do 1 billion search_read ^^

Reply:
`self.env['forum.post'].search_read([('parent_id', 'in', question_ids)], ['id']) `is called one time.